tests/test__io_utils.py

```diff
@@ -30,6 +38,16 @@ def test_read_return_reader(uses_sample_plugin):
     assert reader.command == f"{SAMPLE_PLUGIN_NAME}.some_reader"
 
 
+def test_read_uppercase_extension_return_reader(uses_sample_plugin):
+    # sample plugin hard-codes lower case and returns this error
+    # so the error ensures the matching was case-insensitive
+    # and that the sample plugin received full path (with case)
+    with pytest.raises(ValueError, match="Test plugin should"):
```

Comment:
Where is this exception raised? I see three lines; this construction should only be used for a single line. Based on the code, it looks like lines 47 and 48 are not executed.

Reply:
Yes, sorry copy paste error. I will delete those, but looks like the whole test will be different anyways.